src/main/java/com/r307/arbitrader/service/NotificationServiceImpl.java

```diff
@@ -147,12 +127,92 @@ public void sendEmailNotificationBodyForExitTrade(Spread spread, BigDecimal long
 
         final BigDecimal profit = updatedBalance.subtract(entryBalance);
 
-        final String emailBody = "***** EXIT *****\n" +
+        // This is equivalent of if(isActivePositionExpired) {timeout} else if(isForceExitPosition) {forced} else {exit}
+        final String message = isActivePositionExpired ? "***** TIMEOUT EXIT *****\n" : (isForceExitPosition ? "***** FORCED EXIT *****\n" : "***** EXIT *****\n") +
+            exitSpreadString +
             longCloseString +
             shortCloseString +
             String.format("Combined account balances on entry: $%s\n", entryBalance.toPlainString()) +
             String.format("Profit calculation: $%s - $%s = $%s\n", updatedBalance.toPlainString(), entryBalance.toPlainString(), profit.toPlainString());
 
-        sendEmailNotification(EMAIL_SUBJECT_NEW_EXIT, emailBody);
+        sendNotification(EMAIL_SUBJECT_NEW_EXIT, message);
+    }
+
+    /**
+     * Send the notification via email and/or Telegram.
+     * @param subject the notification title (in case of email)
+     * @param message the notification body/message
+     */
+    @Override
+    public void sendNotification(String subject, String message) {
+        sendInstantMessage(message);
```

Comment:
The main reasons why I set Telegram like email notifications are:

1. to avoid being rate limited (when sending all logs) and to be sure you will always get the the entry and exit trade information. Currently with discord (the one I am using) I am sometimes rate limited and I miss some lines in an entry/exit trade
2. to not have my phone frequently ringing. I can mute the group chat but then I get into the next issue:
3. to have instant notification when a trade occurs and not forcing me to pro actively open discord/email client/slack to check if the bot happen to have entered/exited a trade
4. I am not always interested in the logs, mostly while at work or away from the computer I am only interested in checking whether the bot did a trade or not

With this in mind I thought that Telegram could be a good candidate for only important notifications.

It is possible to send DM on Telegram but the setup is more cumbersome. It evolves the bot having to be able to read incoming messages from the user because (AFAIK) for DM the user has to first send a message to the bot. The bot reads the message and retrieves a `messageId` and replies back to the user with a value for `messageId = messageId + 1` and we must keep track of this messageId for future DM.
In my opinion, using the group way to enable the bot to send messages to the user is more user friendly and much more practical from the code point of view.

Reply:
I also thought in the long term to be able to send commands to the bot via Telegram. Commands like:

- force entry
- force exit
- check account balance
- current trades
- etc